src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -300,6 +300,9 @@ private int AddUniqueTrack(int i, int flags)
         /// <summary>A macro for _ilg.Emit(OpCodes.Newobj, constructor).</summary>
         protected void Newobj(ConstructorInfo constructor) => _ilg!.Emit(OpCodes.Newobj, constructor);
 
+        /// <summary>A macro for _ilg.Emit(OpCodes.Dup).</summary>
+        protected void Dup() => _ilg!.Emit(OpCodes.Dup);
```

Comment:
I thought we were trying to avoid Dup

Reply:
The primary things I wanted to avoid were a) using dup in places the C# compiler wasn't, and b) using dup across branches.  Neither of those is the case here; this is just emitting the same thing Roslyn emits.